### PR TITLE
Improve reliability of podmetric test

### PR DESCRIPTION
**Pull request by ggriffiths**
**What this PR does / why we need it**:
Before, we were assuming that the first callhome interval with data would be in the first 5mins. This first interval would have a fraction of metering data as the pods are started in the middle of an interval. The second interval would then have accurate pod hour data for an entire interval.

However, this is not true in cases where the callhome agent is started before the metering agent with the same interval. In this case, it's possible for the first callhome interval to be empty. The second will be pro-rated callhome data, the 3rd will be the full interval.

To be sure we are hitting an entire callhome interval, we will wait until the 3rd metering.

Additionally, we'll increase the acceptable error rate to 2% as I noticed network slowness resulted in 5m4s intervals last night.

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by ggriffiths** [comment]
Passed with this code
https://jenkins.pwx.dev.purestorage.com/job/Torpedo/job/tp-pxone-podmetric/18/

**Review by dahuang-purestorage — APPROVED** [review]

File: tests/basic/podmetric_test.go [hunk 1]
```diff
@@ -76,40 +76,59 @@ var _ = Describe("{PodMetricFunctional}", func() {
 				ValidateApplications(contexts)
 
 				waitDuration := meteringInterval + 30*time.Second
-				log.InfoD("Wait %v for previous interval to go through", waitDuration)
+				log.InfoD("Wait %v for initial interval to go through in case the metering interval is after the callhome interval", waitDuration)
 				time.Sleep(waitDuration)
 
-				log.InfoD("Wait %v for a new interval to go through", waitDuration)
+				log.InfoD("Wait %v for previous pro-rated interval to go through", waitDuration)
+				time.Sleep(waitDuration)
```
[the remark this answers is not in the document]
Comment (ggriffiths) [reply tests/basic/podmetric_test.go:83]: Yes the sleeps are for different reasons as per the logs surrounding them. We'll keep them separate

File: tests/basic/podmetric_test.go [hunk 2]
```diff
@@ -76,40 +76,59 @@ var _ = Describe("{PodMetricFunctional}", func() {
...
+						return false
+					}
+
+					existsData := len(meteringData) > 0
+					if !existsData {
+						log.Errorf("Failed to get metering data. Retrying...")
+						return false
+					}
+					for _, md := range meteringData {
+						if md.ClusterUUID != clusterUUID {
```
[the remark this answers is not in the document]
Comment (ggriffiths) [reply tests/basic/podmetric_test.go:104]: THis is pulling metering data from one cluster, so every data point will have the same cluster UUID.

File: tests/basic/podmetric_test.go [hunk 3]
```diff
@@ -76,40 +76,59 @@ var _ = Describe("{PodMetricFunctional}", func() {
...
+							log.Errorf("Cluster id does not match. expected: %v actual: %v", clusterUUID, md.ClusterUUID)
+							return false
+						}
+					}
+
+					log.InfoD("Check pod hours is correct")
+					expectedAppPodHours, err := getExpectedPodHours(contexts, meteringInterval)
+					if err != nil {
+						log.Errorf("failed to get expectedAppPodHours: %v. Retrying... %v", err)
```
[the remark this answers is not in the document]
Comment (ggriffiths) [reply tests/basic/podmetric_test.go]: Addressed

**Pull request merged**